discovery/client.go

```diff
@@ -137,7 +137,7 @@ func (r *defaultClientRegistrationManager) findCredentialsAndBuildPresentation(c
 	if err != nil {
 		return nil, fmt.Errorf("failed to match Discovery Service's Presentation Definition (service=%s, did=%s): %w", service.ID, subjectDID, err)
 	}
-	if len(matchingCredentials) == 0 {
+	if len(matchingCredentials) == 0 && len(service.PresentationDefinition.InputDescriptors) > 0 {
```

Comment:
they could be optional? (`min: 0`)

Not sure if we already need to support this, though... Use case?

Reply:
this is the check we do now on multiple locations.